x-pack/packages/kbn-elastic-assistant/impl/connectorland/connector_setup/index.tsx

```diff
@@ -91,24 +89,40 @@ export const useConnectorSetup = ({
   );
 
   // User constants
-  const userName = conversation.theme?.user?.name ?? i18n.CONNECTOR_SETUP_USER_YOU;
-  const assistantName = conversation.theme?.assistant?.name ?? i18n.CONNECTOR_SETUP_USER_ASSISTANT;
+  const userName = useMemo(
+    () => conversation.theme?.user?.name ?? i18n.CONNECTOR_SETUP_USER_YOU,
+    [conversation.theme?.user?.name]
+  );
+  const assistantName = useMemo(
+    () => conversation.theme?.assistant?.name ?? i18n.CONNECTOR_SETUP_USER_ASSISTANT,
+    [conversation.theme?.assistant?.name]
+  );
+  const lastConversationMessageIndex = useMemo(
+    () => conversation.messages.length - 1,
+    [conversation.messages.length]
+  );
 
   const [currentMessageIndex, setCurrentMessageIndex] = useState(
     // If connector is configured or conversation has already been replayed show all messages immediately
     isConnectorConfigured || conversationHasNoPresentationData(conversation)
-      ? MESSAGE_INDEX_BEFORE_CONNECTOR
+      ? lastConversationMessageIndex
       : 0
   );
 
+  const streamingTimeoutRef = useRef<number | undefined>(undefined);
+
   // Once streaming of previous message is complete, proceed to next message
   const onHandleMessageStreamingComplete = useCallback(() => {
-    const timeoutId = setTimeout(() => {
+    if (currentMessageIndex === lastConversationMessageIndex) {
+      clearTimeout(streamingTimeoutRef.current);
+      return;
+    }
+    streamingTimeoutRef.current = window.setTimeout(() => {
       bottomRef.current?.scrollIntoView({ block: 'end' });
       return setCurrentMessageIndex(currentMessageIndex + 1);
     }, conversation.messages[currentMessageIndex]?.presentation?.delay ?? 0);
-    return () => clearTimeout(timeoutId);
-  }, [conversation.messages, currentMessageIndex]);
+    return () => clearTimeout(streamingTimeoutRef.current);
```

Comment:
is this called anywhere?

Reply:
It gets called as `onStreamingComplete()` after being passed into [`StreamingText`](https://github.com/elastic/kibana/blob/6ef069972e8718745478f2b0244fab9200da930c/x-pack/packages/kbn-elastic-assistant/impl/assistant/streaming_text/index.tsx#L27-L43`), but return value isn't typed or used at the moment.

Just ended up using the `useRef` for managing within this file.